ROADMAP.md

```diff
@@ -0,0 +1,123 @@
+  This is both a mid and long term goal we'll be constantly focusing on. See issues [#228](https://github.com/grafana/xk6-browser/issues/228), [#241](https://github.com/grafana/xk6-browser/issues/241) and [#58](https://github.com/grafana/xk6-browser/issues/58).
+
+  *How will we achieve this?*<br>
+  By gradually adding more unit and functional tests and fixing issues when the opportunity arises. We won't have long stints where we focus exclusively on this goal, but it's something we'll dedicate as much time as possible to.
+
+  *Definition of Done*<br>
+  The goal is not to reach 100% code coverage, but, realistically speaking, 80 or 90% should be achievable. Our CI test runs should be stable and flake-free, and we should have no linter issues.
+
+
+Long-term goals
+---------------
+
+These are goals achievable after a year, tentatively done in 2024 and beyond.
+
+- **Add support for Firefox and other WebKit-based browsers.**<br>
+  Currently (April 2022), our main focus is supporting Chromium-based browsers, which should be expanded to include other browsers as well. The main challenges here will be around CDP and the behavior differences between browsers.
+
+  *How will we achieve this?*<br>
+  By testing other browsers and fixing issues as they arise.
+
+  *Definition of Done*<br>
+  When the latest Firefox and certain WebKit-based browsers are as well supported as Chromium.
+
+
+- **Feature parity with Playwright and Puppeteer.**<br>
+  Currently, our functionality is limited compared to more mature projects like Playwright and Puppeteer. We plan to expand this gradually and reach or exceed the features offered by other browser automation tools.
+
+  *How will we achieve this?*<br>
+  By prioritizing new features to add based on API importance and user feedback. After the short-term stability improvements are made, our main focus will be to add more missing features and close the current feature gap.
+
+  *Definition of Done*<br>
+  When we implement all of the functionality found in other tools that makes sense for xk6-browser. This is intentionally vague at the moment, and we'll refine it as we make progress.
+
+
+- **Merge the extension into the main k6 repository.**<br>
+  k6 and Grafana consider browser testing to be another strategy under the testing umbrella, along with performance, contract, and other types of testing modern web applications benefit from. As such, the scope of the k6 tool will expand to include functional/E2E testing using browser automation, with the ultimate goal of merging the xk6-browser extension into the main k6 repository as a core JS module.
+
+  *How will we achieve this?*<br>
+  Once the extension is stable, feature-complete, and widely used in k6 Cloud, we will create a PR to the main k6 repository with the extension code. Then we will deprecate the standalone xk6-browser repository, and point users to use the main k6 binary instead.
```

Comment:
I think this should be a mid-term goal rather than a long-term. I don't think the extension should be "feature-complete" before it's merged. I think the requirements for merging should be:
1. The API is relatively stable (existing scripts should run in a new release) and
2. The code should be relatively bug-free. (doesn't crash or produce invalid results for 99/100 test runs).

Reply:
Sounds good. Let me know if f38387d resolves this.